Gravatar/IImageCache.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Drawing;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+
+namespace Gravatar
+{
+    // TODO image cache should differentiate by image size, as on a system where DPI changes (eg. laptop/monitor screens) the cached images may be innapropriate
+
+    public interface IImageCache
```

Comment:
💭 Overall, the whole separation into two concrete types plus an interface seems unnecessarily complex because we only ever use this in one form. Eliminating the interface and merging the memory cache with the disk cache would likely simplify everything, especially if the linked list approach was used for the MRU.

Reply:
On my own project I would have refactored the code to be completely unrecognisable from where it is now. The interfaces that are there have redundant members and blurred responsibilities.  But I have tried such refactors in the past and have received push back, so I thought I could fire in a quick PR that improves the situation a little bit, get it merged in, and follow up with another. I personally feel the review here is excessively stringent, though I do enjoy the ideas being thrown around.

Reply:
> I personally feel the review here is excessively stringent, though I do enjoy the ideas being thrown around.

Some of the items in this PR aim to establish new code styles for practices in the code base. Such changes get a different level of scrutiny than things like bug fixes or feature implementations that simply follow the existing style.

Reply:
> only ever use this in one form

The Gource plugin only uses the file system caching. It forces images to be downloaded, then copies them into a folder with different names so that Gource can use them.